lib/adapters/file/write.js

```diff
@@ -29,25 +30,95 @@ class WriteFile extends AdapterWrite {
 
         this.filename = options.file;
         this.format = options.format ? options.format : 'json'; // default to json
+
+        this.append = options.append;
+        this.serializerOptions = {};
+        if (options.append && !_.contains(['jsonl', 'csv'], options.format)) {
+            throw new errors.compileError('RT-INVALID-OPTION-COMBINATION',{
+                option: 'append',
+                rule: 'format="csv" or format="jsonl"'
+            });
+        }
+    }
+
+    _readHeaders() {
+        if (this.append && this.format === 'csv') {
+            return fs.accessAsync(this.filename)
+            .then(() => {
+                return new Promise((resolve, reject) => {
+                    // since the file exists lets make sure to let the csv
+                    // serializer to not send the headers
+                    this.serializerOptions.sendHeaders = false;
```

Comment:
I shall figure out what to do then but for this to work @davidvgalbraith you'd have to supply an `ReadableStream` and `WriteableStream` when appending so we can read the first line for CSV and then do the right thing within the serializer ?